node/node.d.ts

```diff
@@ -405,8 +405,18 @@ declare module "buffer" {
 }
 
 declare module "querystring" {
-    export function stringify(obj: any, sep?: string, eq?: string): string;
-    export function parse(str: string, sep?: string, eq?: string, options?: { maxKeys?: number; }): any;
+    export interface StringifyOptions {
+        encodeURIComponent?: Function;
+    }
+
+    export interface ParseOptions {
+        maxKeys?: number;
+        decodeURIComponent?: Function;
+    }
+
+    export function stringify<T>(obj: T, sep?: string, eq?: string, options?: StringifyOptions): string;
+    export function parse(str: string, sep?: string, eq?: string, options?: ParseOptions): any;
+    export function parse<T extends {}>(str: string, sep?: string, eq?: string, options?: ParseOptions): T;
```

Comment:
Hmm... I didn't find any case when the function `parse` returns any type except a type `Object`:
```
> querystring.parse('a=1')
{ a: '1' }
> querystring.parse('a=')
{ a: '' }
> querystring.parse('=1')
{ '': '1' }
> querystring.parse('')
{}
> querystring.parse(1)
{}
```
even if it used with a custom `decodeURIComponent`:
```
> querystring.parse('a=1', null, null, {decodeURIComponent: x => true})
{ true: true }
> querystring.parse('a=1', null, null, {decodeURIComponent: () => {}})
{ undefined: undefined }

```
Are you sure about it?